models/DataObject/Service.php

```diff
@@ -793,7 +793,7 @@ public static function getOptionsForSelectField(string|Concrete $object, ClassDe
                     DataObject\ClassDefinition\Helper\OptionsProviderResolver::MODE_MULTISELECT
                 );
 
-                if (!$definition->useConfiguredOptions() && $optionsProvider instanceof DataObject\ClassDefinition\DynamicOptionsProvider\MultiSelectOptionsProviderInterface) {
+                if (!$definition->useConfiguredOptions() && $definition instanceof ClassDefinition\Data\Multiselect) {
```

Comment:
```suggestion
                if (!$definition->useConfiguredOptions() && $optionsProvider instanceof DataObject\ClassDefinition\DynamicOptionsProvider\SelectOptionsProviderInterface) {
```
this change looks wrong, there is no intention to limit the `$optionsProvider->getOptions` to multiselect, rather that it was the in-common interface for both types
https://github.com/pimcore/pimcore/pull/15871/files#diff-bfb79ec88ffbefbd512bc7bd6d81712984ded16546d19ebb2df2495e697f9247L21